org.lflang/src/org/lflang/generator/ts/TSReactionGenerator.kt

```diff
@@ -180,6 +186,25 @@ class TSReactionGenerator(
         }
     }
 
+    /**
+     * Generate the mutation's effect type as "WritablePort,"
+     * which differs from the reaction's effect type.
+     */
+    private fun generateMutationSignatureElementForPortEffect(effect: VarRef): String {
```

Comment:
I realize that you had to make this distinction because we set the data type up this way in reactor-ts, but it wasn't a very well-though-out decision on my part to implement it that way. It doesn't hurt reactions to be able to inspect the port, so I suggest that instead augment `ReadWrite` with `getPort()` and `MultiReadWrite` with `getPorts` and remove the distinction between reactions and mutations. WDYT?

Reply:
@lhstrh I think it is an excellent idea!
But I have a question; is the `WritablePort` not necessary? If `WritablePort` is deleted, then there will be many problems with implementation. The reason is the WritablePort is class, and the ReadWrite is type.

Reply:
I tried to fix it, but there are some problems.
If we add the `getPort()` in the `ReadWrite`, we should get the `IOPort`. So, the `ReadWrite`'s type should be `Present`. It caused a type error. (The `ReadWrite` is no class)
I think that we can use the `asWritable()`, which is the `IOPort`'s function of returning the `writablePort`.

https://github.com/Wonseo-C/reactor-ts/tree/getport : This is the branch where I worked.

Reply:
@lhstrh :
By discussing with @hokeun, we guessed your purpose of this review is to make the simple code. And we decided to implement it like this -> https://github.com/lf-lang/lingua-franca/pull/1318/commits/74ad4094bc837200e4667eb3864f78b8b9aa326b.
We thought your suggestion could cause another problem because of the type error (detail explained above), so I just modified the `generateReactionSignatureElementForPortEffect` to get different strings if it is a mutation or not.